apps/clock/js/picker/picker.js

```diff
@@ -44,22 +39,18 @@ define(function(require) {
       var values = [];
       var range = setup.pickers[picker].range;
       var isPadded = setup.pickers[picker].isPadded || false;
-      var valueText = setup.pickers[picker].valueText;
-      var textValues = [];
 
       this.nodes[picker] = setup.element.querySelector('.picker-' + picker);
 
       for (var i = range[0]; i <= range[1]; i++) {
-        values.push(isPadded && i < 10 ? '0' + i : '' + i);
-        if (valueText) {
-          textValues.push(_(valueText, { n: i }));
-        }
+        values.push(i);
       }
 
       this.spinners[picker] = new Spinner({
         element: this.nodes[picker],
         values: values,
-        textValues: textValues.length ? textValues : values
+        isPadded: isPadded,
+        l10nId: setup.pickers[picker].l10nId
       });
```

Comment:
After IRC chat with Zibi, it turns out I was mistaking the clock time spinner with the round arrow icon that happens to be also called spinner.
For the clock time spinner/picker, it should NOT be RTL'ed.
if there's a suffix it should look like "p hh:mm" and if there isn't it should be "hh:mm"
Sorry for the confusion!